### PR TITLE
[CALCITE-6533] Division between integer and decimal produces incorrect result

### PR DESCRIPTION
**Pull request by mihaibudiu**
Turns out it's a bug I introduced during the fix for https://issues.apache.org/jira/browse/CALCITE-6464

### CODE REVIEW

File: testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java [hunk 1]
```diff
@@ -2661,6 +2663,19 @@ void checkModOperatorDivByZero(SqlOperatorFixture f) {
...
+              @Override public int getMaxNumericPrecision() {
+                return 28;
+              }
+
+              @Override public int getMaxNumericScale() {
+                return 10;
+              }
+            }));
+    f0.checkScalarExact("95.0 / 100", "DECIMAL(12, 10) NOT NULL", "0.95");
+    f0.checkScalarExact("95 / 100.0", "DECIMAL(17, 6) NOT NULL", "0.95");
```
Comment (caicancai) [inline testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java:2678]: Maybe adding another test would be better such as` 95/100`

File: core/src/main/java/org/apache/calcite/rel/type/RelDataTypeSystem.java [hunk 2]
```diff
@@ -297,6 +297,7 @@ default boolean shouldUseDoubleMultiplication(RelDataTypeFactory typeFactory,
         int six = Math.min(6, getMaxNumericScale());
         int d = p1 - s1 + s2;
         int scale = Math.max(six, s1 + p2 + 1);
+        scale = Math.min(scale, getMaxNumericScale());
```
Comment (caicancai) [inline core/src/main/java/org/apache/calcite/rel/type/RelDataTypeSystem.java:300]: I don't quite understand the rules, but I see some paradoxes.
Assume the following scenario.
```
six = Math.min(6, getMaxNumericScale()) = getMaxNumericScale()
scale = Math.max(six, s1 + p2 + 1) = six = getMaxNumericScale()
scale = Math.min(scale, getMaxNumericScale()) = getMaxNumericScale()
```
Is this a repeated judgment?

File: testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java (hunk 1 above)
Comment (mihaibudiu) [reply testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java:2678]: We are only fixing a bug for decimals, not for integers.

**Comment by mihaibudiu** [comment]
Since this is a very small bugfix for a bug I have introduced, I will merge it without review.

**Pull request merged**